src/logic/operate.js

```diff
@@ -0,0 +1,31 @@
+/* eslint-disable import/extensions */
+import Big from 'big.js';
+
+const operate = (one, two, op) => {
+  const oneBig = new Big(one);
+  const twoBig = new Big(two);
+
+  if (op === '+') {
+    return oneBig.plus(twoBig);
+  }
+
+  if (op === '-') {
+    return oneBig.minus(twoBig);
+  }
+
+  if (op === 'X') {
+    return oneBig.mul(twoBig);
+  }
+
+  if (op === '/' && two !== 0) {
+    return oneBig.div(twoBig).toFixed(2);
+  }
+
+  if (op === '/' && two === 0) {
+    return Infinity;
+  }
```

Comment:
Remember that the `result` prop in `Display` component should be a string so maybe you could convert these to strings here.